golem_messages/message/base.py

```diff
@@ -148,21 +151,23 @@ def _(self, key, value):
                 deserialize_verify,
                 verify_key=slot_name,
                 verify_class=slot_class,
+                allow_none=allow_none,
             )(
                 key, method(self, key, value)
             )
         return _
     return deserialize_slot
 
 
-def verify_slot_list(slot_name, item_class):
+def verify_slot_list(slot_name, item_class, allow_none=False):
     """
     decorator for Message's `deserialize_slot` method
     ensures that the slot identified by `slot_name` is a list of messages with
     the given instance type (provided in `item_class`)
 
     :param slot_name: the name of the slot to verify
     :param item_class: the class to check list items against
+    :param allow_none: whether we're allowing the slot to be empty

```

Comment:
I don't know if you do this or not, but maybe it is worth to add info about default value?